src/components/CategoriesTabs.tsx

```diff
@@ -1,42 +1,45 @@
+import React, { useReducer } from 'react'
 import { Tabs } from '@mui/material'
 import CategoryBtn from './CategoryBtn'
 import { categoriesData } from '~/data/categories'
-import { useReducer } from 'react'
+import { createAction, ActionType, getType } from 'typesafe-actions'
 
-enum ActionType {
-	SET_ACTIVE_CATEGORY = 'SET_ACTIVE_CATEGORY',
-	SET_ACTIVE_SUB_CATEGORY = 'SET_ACTIVE_SUB_CATEGORY',
-}
+const setActiveCategory = createAction('SET_ACTIVE_CATEGORY')<{
+	activeCategory: number
+	activeSubCategory: number
+}>()
+
+const setActiveSubCategory = createAction('SET_ACTIVE_SUB_CATEGORY')<{
+	activeSubCategory: number
+}>()
+
+const actions = { setActiveCategory, setActiveSubCategory }
+
+type CategoriesAction = ActionType<typeof actions>
 
-interface Action {
-	type: 'SET_ACTIVE_CATEGORY' | 'SET_ACTIVE_SUB_CATEGORY'
-	payload: number
-}
 interface State {
 	activeCategory: number
 	activeSubCategory: number | null
 }
 
-const initialState = {
+const initialState: State = {
 	activeCategory: 0,
 	activeSubCategory: null,
 }
 
-const reducer = (state: State, action: Action): State => {
-	const { type, payload } = action
-	switch (type) {
-		case ActionType.SET_ACTIVE_CATEGORY:
+const reducer = (state: State = initialState, action: CategoriesAction): State => {
+	switch (action.type) {
+		case getType(setActiveCategory):
 			return {
 				...state,
-				activeCategory: payload,
-				activeSubCategory: 0,
+				activeCategory: action.payload.activeCategory,
+				activeSubCategory: action.payload.activeSubCategory,
 			}
-		case ActionType.SET_ACTIVE_SUB_CATEGORY:
+		case getType(setActiveSubCategory):
 			return {
 				...state,
-				activeSubCategory: payload,
+				activeSubCategory: action.payload.activeSubCategory,
 			}
-
 		default:
 			return state
 	}
```

Comment:
You can simplify this code by using this syntax.
https://github.com/piotrwitek/typesafe-actions?tab=readme-ov-file#using-createreducer-api-with-type-free-syntax